packages/cli/src/oclif/commands/promote.js

```diff
@@ -35,8 +45,24 @@ class PromoteCommand extends BaseCommand {
     const assumeYes = 'yes' in this.flags;
 
     let shouldContinue;
-    const changelog = await getVersionChangelog(version);
-    if (changelog) {
+    const { changelog, appMetadata, issueMetadata } = await getVersionChangelog(
+      version
+    );
+    if (!changelog) {
+      let error = `${colors.yellow(
+        'Warning!'
+      )} Changelog not found. Please create a CHANGELOG.md file in a format similar to ${colors.cyan(
+        'https://gist.github.com/xavdid/b9ede3565f1188ce339292acc29612b2'
+      )} with user-facing descriptions.`;
+      if (!appMetadata && !issueMetadata) {
+        error += `\nChangelog did not contain any release metadata. Please make sure issue or app data is present in the changelog in the form ${colors.cyan(
+          'APP: <BUGFIX | FEATURE_UPDATE>-<action key>:<action type>'
+        )} (e.x. 'APP: BUGFIX-doStuff:read') or ${colors.cyan(
+          'ISSUE: <BUGFIX | FEATURE_UPDATE>-<issue id>'
+        )} (e.x. 'ISSUE-1234').`;
```

Comment:
What do these categories, "App", and "Issue" mean?
In your example, for `<action type>` "read", this is an internal term, developers shouldn't be expected to use this

Reply:
Are there more integration builder friendly terms that better describe what categories as an `app` metadata and `issue` metadata?

Reply:
I'd suggest we use a more conventional syntax and avoid inventing our own. I've suggested we get inspiration from [GitHub](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue).

"Integration builder friendly terms" means we should use trigger/create/search instead of read/write/search.

So to link to an issue, the user can write `fix #1234` in the changelog. To link to a trigger or an action, they can write `improve trigger/new_card` or `fix create/add_task`.



Here's a complete example changelog I have in mind:

```
## 1.1.0

- Fix a bug in create/add_task #123
- Add a new trigger trigger/new_card
- Update some dependencies #20

Other notes here... create/hello fixes #890
```

The parser would extract these from the changelog:
- `bugfix: create/add_task`
- `bugfix: #123`
- `feature update: trigger/new_card`
- `feature update: #20`
- `bugfix: #890`

Thoughts?

Reply:
I think that works